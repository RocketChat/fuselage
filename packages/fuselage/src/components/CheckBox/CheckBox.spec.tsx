import React from 'react';
import ReactDOM from 'react-dom';

import { CheckBox } from '../..';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CheckBox />, div);
  ReactDOM.unmountComponentAtNode(div);
});
