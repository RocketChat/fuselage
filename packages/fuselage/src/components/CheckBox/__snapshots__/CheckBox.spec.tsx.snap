// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[CheckBox Rendering] renders Checked without crashing 1`] = `
<body>
  <div>
    <label
      class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box"
    >
      <input
        aria-label="Decorator Label"
        checked=""
        class="rcx-box rcx-box--full rcx-check-box__input"
        type="checkbox"
      />
      <i
        aria-hidden="true"
        class="rcx-box rcx-box--full rcx-check-box__fake"
      />
    </label>
  </div>
</body>
`;

exports[`[CheckBox Rendering] renders Default without crashing 1`] = `
<body>
  <div>
    <label
      class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box"
    >
      <input
        aria-label="Decorator Label"
        class="rcx-box rcx-box--full rcx-check-box__input"
        type="checkbox"
      />
      <i
        aria-hidden="true"
        class="rcx-box rcx-box--full rcx-check-box__fake"
      />
    </label>
  </div>
</body>
`;

exports[`[CheckBox Rendering] renders DefaultChecked without crashing 1`] = `
<body>
  <div>
    <label
      class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box"
    >
      <input
        aria-label="Decorator Label"
        checked=""
        class="rcx-box rcx-box--full rcx-check-box__input"
        type="checkbox"
      />
      <i
        aria-hidden="true"
        class="rcx-box rcx-box--full rcx-check-box__fake"
      />
    </label>
  </div>
</body>
`;

exports[`[CheckBox Rendering] renders Disabled without crashing 1`] = `
<body>
  <div>
    <label
      class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box"
    >
      <input
        aria-label="Decorator Label"
        class="rcx-box rcx-box--full rcx-check-box__input"
        disabled=""
        type="checkbox"
      />
      <i
        aria-hidden="true"
        class="rcx-box rcx-box--full rcx-check-box__fake"
      />
    </label>
  </div>
</body>
`;

exports[`[CheckBox Rendering] renders Indeterminate without crashing 1`] = `
<body>
  <div>
    <label
      class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box"
    >
      <input
        aria-label="Decorator Label"
        class="rcx-box rcx-box--full rcx-check-box__input"
        type="checkbox"
      />
      <i
        aria-hidden="true"
        class="rcx-box rcx-box--full rcx-check-box__fake"
      />
    </label>
  </div>
</body>
`;

exports[`[CheckBox Rendering] renders States without crashing 1`] = `
<body>
  <div>
    <table
      class="rcx-box rcx-box--full rcx-css-1s3wr6b"
      style="border-collapse: collapse;"
    >
      <thead
        class="rcx-box rcx-box--full"
      >
        <tr
          class="rcx-box rcx-box--full"
        >
          <td
            aria-hidden="true"
            class="rcx-box rcx-box--full"
          />
          <th
            class="rcx-box rcx-box--full rcx-css-3x4vxb"
          >
            checked
          </th>
          <th
            class="rcx-box rcx-box--full rcx-css-3x4vxb"
          >
            unchecked
          </th>
        </tr>
      </thead>
      <tbody
        class="rcx-box rcx-box--full"
      >
        <tr
          class="rcx-box rcx-box--full"
        >
          <th
            class="rcx-box rcx-box--full rcx-css-3x4vxb"
          >
            default
          </th>
          <td
            class="rcx-box rcx-box--full rcx-css-yoolx6"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-7rd89h"
            >
              <label
                class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box"
              >
                <input
                  aria-label="Decorator Label"
                  checked=""
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </div>
          </td>
          <td
            class="rcx-box rcx-box--full rcx-css-yoolx6"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-7rd89h"
            >
              <label
                class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box"
              >
                <input
                  aria-label="Decorator Label"
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </div>
          </td>
        </tr>
        <tr
          class="rcx-box rcx-box--full"
        >
          <th
            class="rcx-box rcx-box--full rcx-css-3x4vxb"
          >
            hover
          </th>
          <td
            class="rcx-box rcx-box--full rcx-css-yoolx6"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-7rd89h"
            >
              <label
                class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box  is-hovered"
              >
                <input
                  aria-label="Decorator Label"
                  checked=""
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </div>
          </td>
          <td
            class="rcx-box rcx-box--full rcx-css-yoolx6"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-7rd89h"
            >
              <label
                class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box  is-hovered"
              >
                <input
                  aria-label="Decorator Label"
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </div>
          </td>
        </tr>
        <tr
          class="rcx-box rcx-box--full"
        >
          <th
            class="rcx-box rcx-box--full rcx-css-3x4vxb"
          >
            active
          </th>
          <td
            class="rcx-box rcx-box--full rcx-css-yoolx6"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-7rd89h"
            >
              <label
                class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box  is-active"
              >
                <input
                  aria-label="Decorator Label"
                  checked=""
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </div>
          </td>
          <td
            class="rcx-box rcx-box--full rcx-css-yoolx6"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-7rd89h"
            >
              <label
                class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box  is-active"
              >
                <input
                  aria-label="Decorator Label"
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </div>
          </td>
        </tr>
        <tr
          class="rcx-box rcx-box--full"
        >
          <th
            class="rcx-box rcx-box--full rcx-css-3x4vxb"
          >
            focus
          </th>
          <td
            class="rcx-box rcx-box--full rcx-css-yoolx6"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-7rd89h"
            >
              <label
                class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box  is-focused"
              >
                <input
                  aria-label="Decorator Label"
                  checked=""
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </div>
          </td>
          <td
            class="rcx-box rcx-box--full rcx-css-yoolx6"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-7rd89h"
            >
              <label
                class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box  is-focused"
              >
                <input
                  aria-label="Decorator Label"
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </div>
          </td>
        </tr>
        <tr
          class="rcx-box rcx-box--full"
        >
          <th
            class="rcx-box rcx-box--full rcx-css-3x4vxb"
          >
            disabled
          </th>
          <td
            class="rcx-box rcx-box--full rcx-css-yoolx6"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-7rd89h"
            >
              <label
                class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box"
              >
                <input
                  aria-label="Decorator Label"
                  checked=""
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  disabled=""
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </div>
          </td>
          <td
            class="rcx-box rcx-box--full rcx-css-yoolx6"
          >
            <div
              class="rcx-box rcx-box--full rcx-css-7rd89h"
            >
              <label
                class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-check-box"
              >
                <input
                  aria-label="Decorator Label"
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  disabled=""
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;
